subsys/net/lib/aws_iot/src/aws_iot.c

```diff
@@ -0,0 +1,679 @@
+		.utf8 = get_accepted_topic,
+		.size = GET_ACCEPTED_TOPIC_LEN },
+		.qos = MQTT_QOS_1_AT_LEAST_ONCE },
+#endif
+#if defined(CONFIG_AWS_IOT_GET_REJECTED_TOPIC_SUBSCRIBE)
+	{ .topic = {
+		.utf8 = get_rejected_topic,
+		.size = GET_REJECTED_TOPIC_LEN },
+		.qos = MQTT_QOS_1_AT_LEAST_ONCE },
+#endif
+#if defined(CONFIG_AWS_IOT_UPDATE_ACCEPTED_TOPIC_SUBSCRIBE)
+	{ .topic = {
+		.utf8 = update_accepted_topic,
+		.size = UPDATE_ACCEPTED_TOPIC_LEN },
+		.qos = MQTT_QOS_1_AT_LEAST_ONCE },
+#endif
+#if defined(CONFIG_AWS_IOT_UPDATE_REJECTED_TOPIC_SUBSCRIBE)
+	{ .topic = {
+		.utf8 = update_rejected_topic,
+		.size = UPDATE_REJECTED_TOPIC_LEN },
+		.qos = MQTT_QOS_1_AT_LEAST_ONCE },
+#endif
+#if defined(CONFIG_AWS_IOT_UPDATE_DELTA_TOPIC_SUBSCRIBE)
+	{ .topic = {
+		.utf8 = update_delta_topic,
+		.size = UPDATE_DELTA_TOPIC_LEN },
+		.qos = MQTT_QOS_1_AT_LEAST_ONCE },
+#endif
+#if defined(CONFIG_AWS_IOT_DELETE_ACCEPTED_TOPIC_SUBSCRIBE)
+	{ .topic = {
+		.utf8 = delete_accepted_topic,
+		.size = DELETE_ACCEPTED_TOPIC_LEN },
+		.qos = MQTT_QOS_1_AT_LEAST_ONCE },
+#endif
+#if defined(CONFIG_AWS_IOT_DELETE_REJECTED_TOPIC_SUBSCRIBE)
+	{ .topic = {
+		.utf8 = delete_rejected_topic,
+		.size = DELETE_REJECTED_TOPIC_LEN },
+		.qos = MQTT_QOS_1_AT_LEAST_ONCE },
+#endif
+};
+
+struct aws_iot_tx_data {
+	char *buf;
+	size_t len;
+	char *topic;
+	enum mqtt_qos qos;
+};
+
+struct app_topic_data {
+	struct mqtt_topic *list;
+	size_t list_count;
+};
+
+static struct app_topic_data app_topic_data = {.list_count = 0};
+
+static char rx_buffer[CONFIG_AWS_IOT_MQTT_RX_TX_BUFFER_LEN];
+static char tx_buffer[CONFIG_AWS_IOT_MQTT_RX_TX_BUFFER_LEN];
+static char payload_buf[CONFIG_AWS_IOT_MQTT_PAYLOAD_BUFFER_LEN];
+
+static struct mqtt_client client;
+static struct sockaddr_storage broker;
+static struct cloud_backend *aws_iot_backend;
+
+static int client_id_get(char *const id, size_t id_len)
+{
+	int err;
+
+	err = at_cmd_write("AT+CGSN", id, id_len, NULL);
+	if (err) {
+		LOG_ERR("Could not get device IMEI, error: %d", err);
+		return err;
+	}
+
+	id[AWS_CLIENT_ID_LEN] = 0;
+
+	return 0;
+}
+
+static int aws_iot_topics_populate(void)
+{
+	int err;
+
+	err = client_id_get(client_id_buf, sizeof(client_id_buf));
+	if (err) {
+		return err;
+	}
+
+	err = snprintf(shadow_base_topic, sizeof(shadow_base_topic),
+		       SHADOW_BASE_TOPIC, client_id_buf);
+	if (err != SHADOW_BASE_TOPIC_LEN) {
+		return -ENOMEM;
+	}
+
+	err = snprintf(get_topic, sizeof(get_topic), GET_TOPIC, client_id_buf);
+	if (err != GET_TOPIC_LEN) {
+		return -ENOMEM;
+	}
+
+	err = snprintf(update_topic, sizeof(update_topic), UPDATE_TOPIC,
+		       client_id_buf);
+	if (err != UPDATE_TOPIC_LEN) {
+		return -ENOMEM;
+	}
+
+	err = snprintf(delete_topic, sizeof(delete_topic),
+		       DELETE_TOPIC, client_id_buf);
+	if (err != DELETE_TOPIC_LEN) {
+		return -ENOMEM;
+	}
+
+#if defined(CONFIG_AWS_IOT_GET_ACCEPTED_TOPIC_SUBSCRIBE)
+	err = snprintf(get_accepted_topic, sizeof(get_accepted_topic), GET_ACCEPTED_TOPIC,
```

Comment:
This and a few lines below are quite lengthy, remember we have 80 characters/line limit.